database/migrations/2025_03_28_000001_add_ruleset_id_to_tags.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+declare(strict_types=1);
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration
+{
+    public function up(): void
+    {
+        Schema::table('tags', function (Blueprint $table) {
+            $table->smallInteger('ruleset_id')->nullable()->unsigned()->after('name');
+            $table->dropIndex('tags_name_unique');
+            $table->unique(['name', 'ruleset_id'], 'tags_name_ruleset_id_unique');
```

Comment:
rows with null ruleset id won't have the unique constraint...

Reply:
I wonder if this would make it confusing on whether or not there's database level constraint on uniqueness 🤔

Reply:
Apparently not needed since they're inserted manually 🤷